src/pyedb/dotnet/edb_core/cell/hierarchy/component.py

```diff
@@ -808,7 +808,9 @@ def _set_model(self, model):  # pragma: no cover
             return False
         return True
 
-    def assign_spice_model(self, file_path, name=None, sub_circuit_name=None):
+    def assign_spice_model(
+        self, file_path: str, name: str = None, sub_circuit_name: str = None, terminal_pairs: list = None
```

Comment:
If you are adding type hints, you should add optional inputs as `Optional[str]` and `Optional[list]` and adding associated import `from typing import Optional`